src/test/java/uk/gov/ons/ctp/integration/rhsvc/repository/RespondentCaseRepositoryTest.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.ons.ctp.integration.rhsvc.repository;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.ons.ctp.common.FixtureHelper;
+import uk.gov.ons.ctp.common.cloud.RetryableCloudDataStore;
+import uk.gov.ons.ctp.common.event.model.CaseUpdate;
+
+@ExtendWith(MockitoExtension.class)
+public class RespondentCaseRepositoryTest {
+
+  private static final String UPRN = "123456";
+
+  @Spy private RetryableCloudDataStore mockCloudDataStore;
+
+  @InjectMocks private RespondentCaseRepository caseRepo;
+
+  private List<CaseUpdate> collectionCase;
+  private final String[] searchByUprnPath = new String[] {"sample", "uprn"};
+
+  /** Setup tests */
+  @BeforeEach
+  public void setUp() throws Exception {
+    this.collectionCase = FixtureHelper.loadPackageFixtures(CaseUpdate[].class);
+    ReflectionTestUtils.setField(caseRepo, "caseSchema", "SCHEMA");
+  }
+
+  /** Returns Empty List where no valid Address cases are returned from repository */
```

Comment:
consider adding unit tests for getting attributes other than UPRN

Reply:
One of the test cases has been modified to search by postcode.